doltpy/etl/sql_sync/sync_tools.py

```diff
@@ -42,3 +45,53 @@ def _sync_helper(source_reader, target_writer, table_map: Mapping[str, str]):
     to_sync = source_reader(list(table_map.keys()))
     remapped = {table_map[source_table]: source_data for source_table, source_data in to_sync.items()}
     target_writer(remapped)
+
+
+def sync_schema_to_dolt(source_engine: Engine, repo: Dolt, table_map: Mapping[str, str], type_mapping: dict):
+    """
+
+    :param source_engine:
+    :param repo:
+    :param table_map:
+    :param type_mapping:
+    :return:
+    """
+    source_metadata = MetaData(bind=source_engine)
+    source_metadata.reflect()
+    target_metadata = MetaData(bind=repo.engine)
+    target_metadata.reflect()
+    for source_table_name, target_table_name in table_map.items():
+        source_table = source_metadata.tables[source_table_name]
+        target_table = coerce_schema_to_dolt(target_table_name, source_table, type_mapping)
+        if target_table_name in target_metadata.tables.keys():
+            # drop the table if it exists, Dolt will compute diffs when we recreate it
```

Comment:
dropping and adding a table won't recreate the same table with a different schema, you'll just have two tables with the same name and similar schemas. If you want to be able to diff across the changes, you should compute the mutation using sql diff.